receiver/mongodbreceiver/config.go

```diff
@@ -0,0 +1,71 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mongodbreceiver
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"time"
+
+	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/config/configtls"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+	"go.uber.org/multierr"
+)
+
+type Config struct {
+	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
+	configtls.TLSClientSetting              `mapstructure:"tls,omitempty"`
+	Hosts                                   []confignet.NetAddr `mapstructure:"hosts"`
+	Username                                string              `mapstructure:"username"`
+	Password                                string              `mapstructure:"password"`
+	ReplicaSet                              string              `mapstructure:"replica_set,omitempty"`
+	Timeout                                 time.Duration       `mapstructure:"timeout"`
+}
+
+func (c *Config) Validate() error {
+	if len(c.Hosts) == 0 {
+		return errors.New("no hosts were specified in the config")
+	}
+
+	var err error
+	for _, host := range c.Hosts {
+		if host.Transport != "unix" {
```

Comment:
Since we're not doing anything when it is unix, just continue. This keeps the code flatter, which is encouraged in golang.
```suggestion
		if host.Transport == "unix" {
			continue
		}
		
		// continue processing as non-unix
```